File: Software/Chapter 14/cs/Listing14-1/Listing14-1/Listing14-1.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using Microsoft.SqlServer.Server;

namespace APress.Samples
{
    public partial class UserDefinedFunctions
    {
        [Microsoft.SqlServer.Server.SqlFunction(DataAccess=DataAccessKind.None,
            IsDeterministic=true)]
        public static SqlDouble Fahrenheit2Celsius(SqlDouble f)
        {
            SqlDouble c = new SqlDouble();
            c = (5.0 / 9.0) * (f - 32.0);
            return c;
        }
    }
};

File: Software/Chapter 14/vb/Listing14-1/Listing14-1/Listing14-1.vb
Imports System
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports Microsoft.SqlServer.Server

Namespace APress.Samples
    Partial Public Class Sql
        <SqlFunction(DataAccess:=DataAccessKind.None, _
            IsDeterministic:=True)> _
        Public Shared Function Fahrenheit2Celsius(ByVal f As SqlDouble) As SqlDouble
            Dim c As New SqlDouble
            c = (5.0 / 9.0) * (f - 32.0)
            Return c
        End Function
    End Class
End Namespace

